clang/test/SemaCXX/gh67687.cpp

```diff
@@ -0,0 +1,7 @@
+// RUN: %clang_cc1 -verify -fsyntax-only -std=c++17 %s
+
+struct S {
+  int n;
+  int d = (4, []() { return n; }());  // expected-error {{'this' cannot be implicitly captured in this context}} \
+                                      // expected-note {{explicitly capture 'this'}}
+};
```

Comment:
Can you move that test to an existing file? Maybe test/SemaCXX/lambda-implicit-this-capture.cpp
(in a `GH67687` namespace)